### PR TITLE
[FEATURE] Support container name or image

### PR DESCRIPTION
**Pull request by mickaelperrin**

### CODE REVIEW

File: src/Config/ConfigLoader.php [hunk 1]
```diff
@@ -32,12 +33,13 @@ function load(string $workingDir): Config
 
         $data = \array_replace_recursive(self::DEFAULTS, $data);
 
-        $data['image'] !== null
-            or U::fail('Image name is not specified');
+        $data['image'] !== null or $data['container'] !== null
```
Comment (shira-374) [inline src/Config/ConfigLoader.php]:
```suggestion
        $data['image'] !== null || $data['container'] !== null
```

`or` is only used for the error throwing part due to its low operator priority.

File: src/Config/Config.php [hunk 2]
```diff
@@ -10,6 +10,7 @@ class Config
     function __construct(
         public string $baseDir,
         public string $image,
+        public string $container,
```
Comment (shira-374) [inline src/Config/Config.php]: `$image` and `$container` should be nullable.

File: src/Config/ConfigLoader.php [hunk 3]
```diff
@@ -32,12 +33,13 @@ function load(string $workingDir): Config
...
+            or U::fail('Image or container name is not specified');
 
         return new Config(
             $baseDir,
-            $data['image'],
+            $data['image'] ?: '',
```
Comment (shira-374) [inline src/Config/ConfigLoader.php]:
```suggestion
            $data['image'],
```

Shouldn't be needed when `Config` constructor accepts `null`s. Same with `$data['container']`.

File: src/Proxy.php [hunk 4]
```diff
@@ -20,7 +20,7 @@ function run(array $args): int
     {
         $env = $this->getEnv();
         [$args, $inputFile] = $this->processArgs($args);
-        $containerId = $this->getContainerId();
+        $containerId = ($this->config->container !== null) ? $this->config->container : $this->getContainerId();
```
Comment (shira-374) [inline src/Proxy.php]:
```suggestion
        $containerId = $this->config->container ?? $this->getContainerId();
```

**Review by shira-374 — CHANGES_REQUESTED** [review]

**Comment by mickaelperrin** [comment]
Thanks for taking time to comment my PR. I guess this is ok now.

**Comment by shira-374** [comment]
Thanks

**Pull request merged**